### DISCUSSION TITLE
Node Configuration

### DISCUSSION THREAD

**Discussion by MightyTiamat** [jensrossbach/node-red-contrib-chronos#119] [Q&A] [answered]
Hi There, I'm trying to use the **Time Change Node** to update the time to a particular time zone instead of my local time to route based on the time of day in that location.  I believe that I have configured everything correctly from the instructions, but the time is still being output as the location node red is installed (cant remember the exact term).

Node = Time Change
Configuration = I've added a configuration for Spain using the information available via https://nodatime.org/TimeZones.
          Latitude = 40.400
          Longitude = -3.683
          Time Zone = Europe/Madrid

![2](https://github.com/jensrossbach/node-red-contrib-chronos/assets/157701963/808e228c-a32a-4621-97b0-b5660fb93241)

Change Rules = I've used Set = **msg.time** to save the time to be used later in the flow with the To set as **current time**.

![1](https://github.com/jensrossbach/node-red-contrib-chronos/assets/157701963/bdcc5ac8-1266-41eb-8f20-8931f2299b59)

Once configured I'm expecting this to take the time from the time zone configuration and then output that as msg.time.  However the reality is that the time output is my local time in the UK instead of the time zone configured for Spain.

**Comment by jensrossbach** [comment]
Hi, actually the current time is stored as a Unix epoch timestamp in the message property. Such timestamps are always in UTC and do not contain any time zone information. If you display that timestamp in the debug panel, Node-RED will display it in your local time zone as it does not have any information about the time zone you configured in the Chronos config. My proposal would be to add another rule which converts the timestamp to an ISO 8601 string including the time zone information:

![image](https://github.com/jensrossbach/node-red-contrib-chronos/assets/5118924/ce33a351-6fbf-4766-8fbe-af8497df4157)

Unfortunately there is currently no easier way to specify an ISO 8601 format. Maybe in one of the next versions I will add the possibility to supply an empty format string for the convert rule which will then create an ISO 8601 string.

**Reply by MightyTiamat** [reply]
Hi There, I don't seem to have the same options as yourself on the version of the nodes pack installed on my pallet 1.19.0 or 1.19.1.  I only see the below options.
 
![3](https://github.com/jensrossbach/node-red-contrib-chronos/assets/157701963/6b4bf8a8-46d9-46b6-bfca-c4ccc1399e00)

I do however have the option on an older node pack 1.18.3, but the problem I have is I don't have this version on the Dialogue Studio that I have to use to write this particular flow.

I can use change.list.changeType.toString, but doing that removes all the time properties meaning that I can't use Time Switch nodes to route things based on the options available.  I can manipulate the string to pull out the information I need using the Moment Nodes, but that makes things a little more long winded, but does allow me to route using the regular Switch Nodes.

Hopefully this all makes sense.

**Reply by jensrossbach** [reply, marked as answer]
> I don't seem to have the same options as yourself on the version of the nodes pack installed on my pallet 1.19.0 or 1.19.1. I only see the below options.

This is strange, I cannot reproduce this issue with the untranslated string on my side. Anyway, `change.list.changeType.toString` refers to the same rule that I mentioned (should normally be displayed as "Convert").

> I can use change.list.changeType.toString, but doing that removes all the time properties meaning that I can't use Time Switch nodes to route things based on the options available.

The convert rule only changes the property that you specify in the rule. If you need it in two forms, you can also create two properties and only convert one of them to a string.

Anyway, if I understood you correctly, you want to route based on current local time in Spain although your Node-RED instance is in UK and has the UK time zone, correct?

What about creating the current time property with a change node as you did already (you don't need to configure the time change node for Spain time zone as the timestamp is UTC as I described above). Then you feed it to a time switch node which uses a configuration with Spain time zone. This way, the time switch node should interpret the timestamp in Spain local time instead of UK local time.

**Reply by MightyTiamat** [reply]
Hi There,  I really appreciate your support on this problem I was having with this node.  I am now able to do everything that I want.  It appears that is was my mistake in the way I was configuring the Time Switch Nodes, but the error in the drop down wasn't helping.  Now that I understand what was happening in the Time Change and how to configure that correctly along with the Time Switch Node, I'm able to get everything to behave as I was expecting.

Thank you for your help as this all really helps me along my journey.
